cmd/system-probe/config/config.go

```diff
@@ -195,6 +199,16 @@ func (c Config) ModuleIsEnabled(modName ModuleName) bool {
 	return ok
 }
 
+// RequiresEBPF returns a bool indicating if at least one enabled module requires eBPF.
+func (c Config) RequiresEBPF() bool {
```

Comment:
I think this loop could be `cmd/system-probe/api/module.Register` before the `preRegister` call, and passed as an additional argument to `preRegister` and `postRegister`.